pkg/authorization/admission/restrictusers/restrictusers.go

```diff
@@ -0,0 +1,248 @@
+package restrictusers
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	oclient "github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	usercache "github.com/openshift/origin/pkg/user/cache"
+)
+
+func init() {
+	admission.RegisterPlugin("RestrictSubjectBindings",
+		func(kclient kclientset.Interface, config io.Reader) (admission.Interface,
+			error) {
+			return NewRestrictUsersAdmission(kclient)
+		})
+}
+
+// restrictUsersAdmission implements admission.Interface and enforces
+// restrictions on adding rolebindings in a project to permit only designated
+// subjects.
+type restrictUsersAdmission struct {
+	*admission.Handler
+
+	oclient    oclient.Interface
+	kclient    kclientset.Interface
+	groupCache *usercache.GroupCache
+}
+
+var _ = oadmission.WantsOpenshiftClient(&restrictUsersAdmission{})
+var _ = oadmission.WantsGroupCache(&restrictUsersAdmission{})
+var _ = oadmission.Validator(&restrictUsersAdmission{})
+
+// NewRestrictUsersAdmission configures an admission plugin that enforces
+// restrictions on adding role bindings in a project.
+func NewRestrictUsersAdmission(kclient kclientset.Interface) (admission.Interface, error) {
+	return &restrictUsersAdmission{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+		kclient: kclient,
+	}, nil
+}
+
+func (q *restrictUsersAdmission) SetOpenshiftClient(c oclient.Interface) {
+	q.oclient = c
+}
+
+func (q *restrictUsersAdmission) SetGroupCache(c *usercache.GroupCache) {
+	q.groupCache = c
+}
+
+// objectReferenceDelta returns the relative complement of []ObjectReference a
+// in []ObjectReference b (i.e., a∖b).
+func objectReferenceDelta(a, b []kapi.ObjectReference) []kapi.ObjectReference {
```

Comment:
> Would "left" and "right" be acceptable? Or "lhs" and "rhs"?

Not really.  This is a test of "set of authority" and "things to check against the set of the authority" as opposed to something like a comparison.